contracts/StakeDepositContract.sol

```diff
@@ -108,14 +108,15 @@ contract StakeDepositContract is IDepositContract, IERC165, IERC677Receiver, EIP
         if (count > 1) {
             require(count <= 128, "BatchDeposit: You can deposit max 128 validators at a time");
             require(stake_amount == 32 ether * count, "BatchDeposit: batch deposits require 32 STAKE deposit amount");
+            stake_amount = 32 ether;
```

Comment:
in order to make the code clearer consider to introduce the following construction:
```
uint256 stake_amount_per_deposit = stake_amount
if (count > 1 ) {
    . . .
    stake_amount_per_deposit = 32 ether
}
. . .
for (uint256 p = 32; p < data.length; p += 176) {
    . . .
    _deposit(pubkey, withdrawal_credentials, signature, deposit_data_root, stake_amount_per_deposit);
}
```

Reply:
Sounds reasonable. Fixed.